### PR TITLE
Docker And Docker-Compose Support

### PR DESCRIPTION
**Pull request by ayoubfletcher**
There was a problem with the previous pull request I synced the fork and it forced the pull request to get close my bad.
The previous pull request has some issues that have been pointed out by karlderkaefer the suggested solution didn't work that well forcing the container to run using the current user id and group caused some issues with the code inside the container. But it has been fixed by forcing the owner to nobody after the code generated and most of the generated code seems to prefer Debian distro, so container is using the debian with sudo to run the generated code inside the container.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,6 +62,30 @@ To **run in the browser** you can simply:
...
+
+**Running using docker cli**:
+
+Building the image:
+- `git clone https://github.com/AntonOsika/gpt-engineer.git`
+- `cd gpt-engineer`
+- `docker build --rm -t gpt-engineer .`
+
+Running the container:
+- `docker run -it --rm -e OPENAI_API_KEY="YOUR OPENAI KEY" -v ./wallpheaven-generator:/project gpt-engineer`
```
Comment (karlderkaefer) [inline README.md:75]:
```suggestion
- `docker run -it --rm -e OPENAI_API_KEY="YOUR OPENAI KEY" -v ./your-project:/project gpt-engineer`
```

**Comment by karlderkaefer** [comment]
I won't be able to test it out within next 2 weeks but the code looks good from my side

**Review by karlderkaefer — APPROVED** [review]
works for me 👍

**Comment by AntonOsika** [comment]
Good job @ayoubfletcher and @karlderkaefer 

Keep the good work up .

**Pull request merged**